services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/system_monitor/_disk_usage.py

```diff
@@ -17,41 +21,177 @@
+    _incoming_overwrite_usage: dict[str, DiskUsage] = field(default_factory=dict)
+    _last_usage: dict[str, DiskUsage] = field(default_factory=dict)
+
+    @cached_property
+    def _flat_monitored_paths(self) -> set[Path]:
+        return set.union(*self.monitored_paths.values())
+
+    @cached_property
+    def _normalized_monitored_paths(self) -> dict[MountPathCategory, set[str]]:
+        return {
+            k: {
+                _get_normalized_folder_name(
+                    get_relative_path(p, self.dy_volumes_mount_dir)
+                )
+                for p in paths
+            }
+            for k, paths in self.monitored_paths.items()
+        }
+
+    async def _get_measured_disk_usage(self) -> list[DiskUsage]:
+        return await logged_gather(
+            *[
+                get_usage(monitored_path)
+                for monitored_path in self._flat_monitored_paths
+            ]
+        )
 
-    async def _publish_disk_usage(self, usage: dict[Path, DiskUsage]):
+    def _get_normalized_disk_usage(
+        self, measured_disk_usage: list[DiskUsage]
+    ) -> dict[str, DiskUsage]:
+        return {
+            _get_normalized_folder_name(
+                get_relative_path(p, self.dy_volumes_mount_dir)
+            ): u
+            for p, u in zip(
+                self._flat_monitored_paths, measured_disk_usage, strict=True
+            )
+        }
+
+    def _overwrite_with_incoming_disk_usage(
+        self, normalized_disk_usage: dict[str, DiskUsage]
+    ) -> None:
+        # overwrite disk usage with incoming usage from EFS
+        for key, overwrite_usage in self._incoming_overwrite_usage.items():
+            normalized_disk_usage[key] = overwrite_usage  # noqa: PERF403
+
+    @staticmethod
+    def _get_grouped_usage_to_folder_names(
+        normalized_disk_usage: dict[str, DiskUsage]
+    ) -> dict[DiskUsage, set[str]]:
+        """Groups all paths that have the same metrics together"""
+        usage_to_folder_names: dict[DiskUsage, set[str]] = {}
+        for folder_name, disk_usage in normalized_disk_usage.items():
+            if disk_usage not in usage_to_folder_names:
+                usage_to_folder_names[disk_usage] = set()
+
+            usage_to_folder_names[disk_usage].add(folder_name)
+        return usage_to_folder_names
+
+    async def _publish_disk_usage(self, usage: dict[str, DiskUsage]):
         await publish_disk_usage(
             self.app, user_id=self.user_id, node_id=self.node_id, usage=usage
         )
 
     async def _monitor(self) -> None:
-        disk_usages: list[DiskUsage] = await logged_gather(
-            *[get_usage(monitored_path) for monitored_path in self.monitored_paths]
-        )
+        measured_disk_usage = await self._get_measured_disk_usage()
+
+        normalized_disk_usage = self._get_normalized_disk_usage(measured_disk_usage)
+
+        self._overwrite_with_incoming_disk_usage(normalized_disk_usage)
```

Comment:
Overwriting the normalized version of the "measured disk usage" with "incoming disk usage" is not clear to me.

Reply:
I've renamed this part, please have a look at it again.